src/Lecture1.hs

```diff
@@ -123,4 +145,17 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = show n ++ " is greater than " ++ show g ++ " elements and lower than " ++ show s ++ " elements"
+    where
+        countCases :: (Int -> Int -> Bool) -> Int -> [Int] -> Int
+        countCases func num lst = go func num 0 lst
+            where
+                go :: (Int -> Int -> Bool) -> Int -> Int -> [Int] -> Int
+                go f x result l
+                  | null l = result
+                  | f x (head l) = go f x (result + 1) (tail l)
+                  | otherwise = go f x result (tail l)
+        g = countCases (>) n list
+        s = countCases (<) n list
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.